runtime/common/src/signature.rs

```diff
@@ -0,0 +1,53 @@
+use common_primitives::utils::wrap_binary_data;
+use sp_runtime::{traits::Verify, AccountId32, MultiSignature};
+use sp_std::vec::Vec;
+
+pub fn check_signature(signature: &MultiSignature, signer: AccountId32, payload: Vec<u8>) -> bool {
```

Comment:
I'm curious, why have this return `bool` instead of following the original pattern of `verify_signature` and return `DispatchResult`? Doing it this way requires more context every place it's called (each call to `ensure!(check_signature(...), Error::?)` requires passing in the error to be thrown, but since it's always `InvalidSignature`, why not just throw that from here and call as `verify_signature(...)?` in the pallet methods?

Reply:
Good question! My reasoning behind returning bool is to give each pallet the flexibility to be more precise with the error handling. For instance, in the MSA pallet, there are multiple signature checks, and it’s useful to provide more specific error messages depending on which check fails. This approach allows us to give the user more detailed feedback, rather than defaulting to a generic InvalidSignature error.